plugins/statistics.koplugin/calendarview.lua

```diff
@@ -719,11 +1237,18 @@ function CalendarView:_populateItems()
             read_books = books_by_day[day_s],
             show_parent = self,
             callback = not is_future and function()
-                -- Just as ReaderStatistics:callbackDaily(), but without any window stacking
-                UIManager:show(KeyValuePage:new{
+                local kv_pairs = self.reader_statistics:getBooksFromPeriod(day_ts, day_ts + 86400)
+                local seconds_books = self.reader_statistics:getReadingDurationBySecond(day_s)
+                for _, kv in ipairs(kv_pairs) do
+                    if seconds_books[kv.book_id] then
+                        kv.periods = seconds_books[kv.book_id].periods
+                    end
+                    kv.checked = true
```

Comment:
Could it happen some books get to be unchecked ?

Reply:
I had thought about only checking the top couple ones but it turned out for me that the others have barely any bars so turning them on/off doesn't make much difference.